### PR TITLE
PLAT-119501: Fixed year picker width to not changing

### PR DESCRIPTION
**Pull request by MikyungKim**
This PR fixes changed year picker width when the value is changing.

Enact-DCO-1.0-Signed-off-by: Mikyung Kim (mikyung27.kim@lge.com)

### CODE REVIEW

File: internal/Picker/Picker.js [hunk 1]
```diff
@@ -203,7 +203,24 @@ const PickerBase = kind({
...
+		 *
+		 * You may also supply a number. This number will determine the minimum size of the Picker.
+		 * Setting a number to less than the number of characters in your longest value may produce
+		 * unexpected results.
+		 *
+		 * @type {('small'|'medium'|'large'|Number)}
+		 * @public
+		 */
+		width: PropTypes.oneOfType([
+			PropTypes.oneOf([null, 'small', 'medium', 'large']),
```
Comment (ybsung) [inline internal/Picker/Picker.js:221]: Are there any code to handle `'small', 'medium', 'large'` values in Agate Pickers like Sandstone? If not, let's support numbers. Then we might fix the qa-a11y Picker view using width prop of `large` and `medium`.

```
	determineClasses (css, decrementerDisabled, incrementerDisabled) {
		...

		return classnames(
			css.picker,
			css[orientation],
			css[width],
```

File: internal/Picker/Picker.js [hunk 2]
```diff
@@ -296,8 +314,14 @@ const PickerBase = kind({
 		const decrementAriaLabel = `${currentValueText} ${decAriaLabel}`;
 		const incrementAriaLabel = `${currentValueText} ${incAriaLabel}`;
 
+		let sizingPlaceholder = null;
+		if (typeof width === 'number' && width > 0) {
+			sizingPlaceholder = <div aria-hidden className={css.sizingPlaceholder}>{'0'.repeat(width)}</div>;
+		}
```
Comment (ybsung) [inline internal/Picker/Picker.js:320]: How about 
```
const sizingPlaceholder = (typeof width === 'number' && width > 0) ?
	<div aria-hidden className={css.sizingPlaceholder}>{'0'.repeat(width)}</div> :
	null;
```
?

File: internal/Picker/Picker.module.less [hunk 3]
```diff
@@ -26,6 +25,14 @@
 		margin: @agate-picker-margin;
 		padding: @agate-picker-padding;
 
+		.sizingPlaceholder {
+			height: 0;
+			visibility: hidden;
+			padding-left: .extract(@agate-picker-padding, left)[];
+			padding-right: .extract(@agate-picker-padding, right)[];
```
Comment (ybsung) [inline internal/Picker/Picker.module.less:32]: image
The year text has no padding.

image
But the `sizingPlaceholder` has the paddings. So the placeholder size will be bigger than the year text. Did you intend to do it?

File: internal/Picker/Picker.module.less [hunk 4]
```diff
@@ -26,6 +25,14 @@
 		margin: @agate-picker-margin;
 		padding: @agate-picker-padding;
 
+		.sizingPlaceholder {
```
Comment (ybsung) [inline internal/Picker/Picker.module.less:28]: image

`backgounrd-image` is applied to `.sizingPlaceholder `. Could you remove it?

**Review by ybsung — CHANGES_REQUESTED** [review]
Could you find my comments?

In addition, it's out of scope, but if you can, could you delete `locale` and `value` props in the `DateTimeBase`?
```
const DateTimeBase = kind({
	name: 'DateTime',

	...
	render: ({children, css, ...rest}) => {
		delete rest.locale;
		delete rest.value;

		return (

			<div {...rest}>
				<div className={css.pickers}>
					{children}
				</div>
			</div>
		);
	}
});
```

File: internal/Picker/Picker.js (hunk 1 above)
Comment (MikyungKim) [reply internal/Picker/Picker.js:221]: We will support those values once animation related code got in soon.

File: internal/Picker/Picker.js (hunk 2 above)
Comment (MikyungKim) [reply internal/Picker/Picker.js:320]: Let's keep the original one.

File: internal/Picker/Picker.module.less (hunk 4 above)
Comment (MikyungKim) [reply internal/Picker/Picker.module.less:28]: 🆗

File: internal/Picker/Picker.module.less (hunk 3 above)
Comment (MikyungKim) [reply internal/Picker/Picker.module.less:32]: The width is the same as 90 + 12 +12 = 114.

**Review by ybsung — APPROVED** [review]
LGTM

**Comment by jeonghee27** [comment]
image
the layout is broken. DId i do something wrong :(

**Comment by MikyungKim** [comment]
> image
> the layout is broken. DId i do something wrong? :(

This happened because the width of DateTimePicker is a fixed value. We need to refactor Picker layout like sandstone's.
It will be handled in the picker animation ticket.

**Pull request merged**